explore/explore.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+namespace exploration {
```

Comment:
This is good.  Maybe add some comments about the meaning of arguments?

Reply:
done

Reply:
The comments look good here.